zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -901,12 +911,20 @@ impl DiskDb {
                 db
             }
 
-            // TODO: provide a different hint if the disk is full, see #1623
-            Err(e) => panic!(
-                "Opening database {path:?} failed: {e:?}. \
-                 Hint: Check if another zebrad process is running. \
-                 Try changing the state cache_dir in the Zebra config.",
-            ),
+            Err(e) => {
+                if e.kind() == rocksdb::ErrorKind::Busy || e.kind() == rocksdb::ErrorKind::IOError {
+                    panic!(
+                        "Database already open {path:?} \
+                         Hint: Check if another zebrad process is running."
+                    )
+                } else {
+                    panic!(
+                        "Opening database {path:?} failed. \
+                        Hint: Try changing the state cache_dir in the Zebra config. \
+                        Error: {e}",
+                    )
+                }
+            }
```

Comment:
Nitpick/Optional/Subjective: I think it looks a little nicer with a `matches!` expr in a match arm guard.

I think it should say "database _likely_ already open" since it's not always the cause of those errors.

```suggestion
            Err(e) if matches!(e.kind(), ErrorKind::Busy | ErrorKind::IOError) => panic!(
                "Database likely already open {path:?} \
                         Hint: Check if another zebrad process is running."
            ),

            Err(e) => panic!(
                "Opening database {path:?} failed. \
                        Hint: Try changing the state cache_dir in the Zebra config. \
                        Error: {e}",
            ),
```